### DISCUSSION TITLE
How to get Alert State via golang

### DISCUSSION THREAD

**Discussion by BrucePeng92** [prometheus-operator/prometheus-operator#5088] [Q&A] [answered]
Can I get alert with state in golang? Like curl `/api/v1/rules?type=alert`?
I used `versioned.Clientset` to connect prometheus.

**Comment by slashpai** [comment, marked as answer]
I suppose you are asking how to query prometheus using api if so you can refer [prometheus api doc ](https://prometheus.io/docs/prometheus/latest/querying/api/#alerts)

**Reply by BrucePeng92** [reply]
I want to get alert rules with state use go-SDK. If prometheus SDK does not support this function,  I will use http request .

**Reply by slashpai** [reply]
That would be a question for [prometheus](https://github.com/prometheus/prometheus). AFAIK the way to do is through API
